SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -801,4 +807,109 @@ void FLocalDeploymentManager::SetAutoDeploy(bool bInAutoDeploy)
+				UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed to trigger snapshot at '%s'; received '%s'"),
+					   *HttpRequest->GetURL(), *HttpResponse->GetContentAsString());
+				if (OnSnapshotTaken != nullptr)
+				{
+					OnSnapshotTaken(false, FString());
+				}
+				return;
+			}
+
+			// Unfortunately by the time this callback happens, the files haven't been flushed, so if you copy you may get
+			// the wrong info! So let's wait a bit..
+			FTimerHandle TimerHandle;
+			World->GetTimerManager().SetTimer(
+				TimerHandle,
+				[bUseStandard, OnSnapshotTaken]() {
+					bool bSuccess = false;
+
+					FString NewestSnapshotFilePath;
+					FString SnapshotSearchDirectory;
+
+					if (bUseStandard)
+					{
+						FSpatialGDKServicesModule& GDKServices =
+							FModuleManager::GetModuleChecked<FSpatialGDKServicesModule>("SpatialGDKServices");
+						FLocalDeploymentManager* LocalDeploymentManager = GDKServices.GetLocalDeploymentManager();
+
+						FString SpatialPath = FPaths::ProjectDir() + TEXT("../spatial/");
+						SnapshotSearchDirectory =
+							SpatialPath + TEXT("logs/localdeployment/") + LocalDeploymentManager->GetLocalRunningDeploymentID();
+
+						IFileManager& FileManager = FFileManagerGeneric::Get();
+
+						TArray<FString> SnapshotFiles;
+
+						FileManager.FindFiles(SnapshotFiles, *SnapshotSearchDirectory, TEXT("snapshot"));
+
+						FDateTime NewestSnapshotTimestamp = FDateTime::MinValue();
+
+						for (const FString& SnapshotFile : SnapshotFiles)
+						{
+							FString SnapshotFilePath = SnapshotSearchDirectory + TEXT("/") + SnapshotFile;
+							FDateTime SnapshotFileTimestamp = FileManager.GetTimeStamp(*SnapshotFilePath);
+							if (SnapshotFileTimestamp > NewestSnapshotTimestamp)
+							{
+								NewestSnapshotTimestamp = SnapshotFileTimestamp;
+								NewestSnapshotFilePath = SnapshotFilePath;
+							}
+						}
+					}
+					else
+					{
+						FString AppDataLocalPath = FPlatformMisc::GetEnvironmentVariable(TEXT("LOCALAPPDATA"));
+						SnapshotSearchDirectory = FString::Printf(TEXT("%s/.improbable/local_snapshots"), *AppDataLocalPath);
+						FString LatestSnapshotInfoPath = FString::Printf(TEXT("%s/latest"), *SnapshotSearchDirectory);
+						FString LatestSnapshot;
+						if (FPaths::FileExists(LatestSnapshotInfoPath)
+							&& FFileHelper::LoadFileToString(LatestSnapshot, *LatestSnapshotInfoPath))
+						{
+							NewestSnapshotFilePath =
+								FString::Printf(TEXT("%s/.improbable/local_snapshots/%s"), *AppDataLocalPath, *LatestSnapshot);
+						}
+					}
+
+					bSuccess = !NewestSnapshotFilePath.IsEmpty();
+
+					if (!bSuccess)
+					{
+						UE_LOG(LogSpatialDeploymentManager, Error, TEXT("Failed find snapshot file in '%s'"), *SnapshotSearchDirectory);
+					}
+
+					if (OnSnapshotTaken != nullptr)
+					{
+						OnSnapshotTaken(bSuccess, NewestSnapshotFilePath);
+					}
+				},
+				0.5f, false);
+		});
+
+	if (bUseStandard)
+	{
+		HttpRequest->SetURL(TEXT("http://localhost:5006/snapshot"));
```

Comment:
One thing to note is that in the future we will switch to using the runtime in `Standalone Mode` which means that these endpoint ports will be configurable and the defaults might change. Perhaps we want to turn these ports into const vars to make it easier to switch in the future.

Reply:
They are just in that place, and we'll have to change the code anyway once those options come, so I'd say we tackle that when it comes.